app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_setting_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(environment, content_path)
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(environment, content_path)
+      return "#{content_path}/#{::Katello::Glue::Pulp::Repos.repo_path_from_content_path(environment, '')}" if
+        content_path.start_with?('http', 'https', 'ftp', 'ftps')
+      return "#{foreman_setting_url}/#{path}" unless @host && @host.content_source
```

Comment:
The repository_url should expect the `@host` to be just present, and fail immediately in case `@host` ins not available, as the contract of the mothod should be that it converts the `content_path` to full url based on `@host` attributes, and without `@host`, it can't provide much of a value to the user.

Reply:
I understand now, it is used on a `@host` template, so if it is not present, it is a bug.